### PR TITLE
amt: Add a security attestation for provisioning

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/amt/fu-plugin-amt.c [hunk 1]
```diff
@@ -544,15 +553,35 @@ void
...
+
+	if (!fu_common_is_cpu_intel ())
+		return;
+
+	attr = fwupd_security_attr_new ("com.intel.AMT");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_SYSTEM_PROTECTION);
+	fwupd_security_attr_set_name (attr, "Intel AMT");
+	fu_security_attrs_append (attrs, attr);
+	if (data->provisioned)
+		fwupd_security_attr_set_result (attr, "provisioned");
```
Comment (hughsie) [inline plugins/amt/fu-plugin-amt.c:584]: i think just `return` here and match the style of the other plugins.

File: plugins/amt/fu-plugin-amt.c [hunk 2]
```diff
@@ -544,15 +553,35 @@ void
...
+
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	if (!fu_common_is_cpu_intel ())
+		return;
+
```
Comment (hughsie) [inline plugins/amt/fu-plugin-amt.c:578]: Don't we need to keep track if there are any AMT devices detected too? remember this vfunc gets called even if fu_plugin_amt_create_device() never got called.

File: plugins/amt/fu-plugin-amt.c (hunk 1 above)
Comment (hughsie) [inline plugins/amt/fu-plugin-amt.c:584]: I think the result needs to be sentance case, e.g. `Provisioned`

Comment (superm1) [reply plugins/amt/fu-plugin-amt.c]: fixed

File: plugins/amt/fu-plugin-amt.c (hunk 2 above)
Comment (superm1) [reply plugins/amt/fu-plugin-amt.c:580]: yes, fixed

File: plugins/amt/fu-plugin-amt.c (hunk 1 above)
Comment (superm1) [reply plugins/amt/fu-plugin-amt.c]: fixed

**Pull request merged**